### ISSUE TITLE
Type Error on new Vite projects with svelte-ts project

### ISSUE THREAD

**Issue by trbernstein** [sveltejs/vite-plugin-svelte#1055]
### Describe the bug

Starting a new Vite project and selecting svelte-ts template (by first selecting svelte and then TypeScript) results in an error, because `code` var in error.js is null.

### Reproduction URL

https://github.com/vitejs/vite/tree/main/packages/create-vite/template-svelte-ts

### Reproduction

1) Start a new Vite project using `npm create vite@latest`
2) Install dependencies in project folder with `npm install`
3) Start dev server with `npm run dev`
4) Observe error.

### Logs

```shell
[vite] Internal server error: Cannot read properties of undefined (reading 'startsWith')
Plugin: vite-plugin-svelte
File: /***/src/App.svelte
    at couldBeFixedByCssPreprocessor (file://***/node_modules/.pnpm/@sveltejs+vite-plugin-svelte@5.0.2_svelte@5.14.2_vite@6.0.3/node_modules/@sveltejs/vite-plugin-svelte/src/utils/error.js:111:72)
    at enhanceCompileError (file://***/node_modules/.pnpm/@sveltejs+vite-plugin-svelte@5.0.2_svelte@5.14.2_vite@6.0.3/node_modules/@sveltejs/vite-plugin-svelte/src/utils/error.js:126:6)
    at compileSvelte (file:/***/node_modules/.pnpm/@sveltejs+vite-plugin-svelte@5.0.2_svelte@5.14.2_vite@6.0.3/node_modules/@sveltejs/vite-plugin-svelte/src/utils/compile.js:148:4)
    at async TransformPluginContext.transform (file:/***/node_modules/.pnpm/@sveltejs+vite-plugin-svelte@5.0.2_svelte@5.14.2_vite@6.0.3/node_modules/@sveltejs/vite-plugin-svelte/src/index.js:164:20)
    at async EnvironmentPluginContainer.transform (file:/***/node_modules/.pnpm/vite@6.0.3/node_modules/vite/dist/node/chunks/dep-yUJfKD1i.js:47001:18)
    at async loadAndTransform (file:/***/node_modules/.pnpm/vite@6.0.3/node_modules/vite/dist/node/chunks/dep-yUJfKD1i.js:40852:27)
    at async viteTransformMiddleware (file:/***/node_modules/.pnpm/vite@6.0.3/node_modules/vite/dist/node/chunks/dep-yUJfKD1i.js:42296:24)
```

### System Info

```shell
System:
    OS: macOS 15.2
    CPU: (24) arm64 Apple M2 Ultra
    Memory: 40.07 GB / 192.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 23.4.0 - /opt/homebrew/bin/node
    npm: 10.9.2 - /opt/homebrew/bin/npm
    pnpm: 9.15.0 - /opt/homebrew/bin/pnpm
  Browsers:
    Safari: 18.2
  npmPackages:
    @sveltejs/vite-plugin-svelte: ^5.0.2 => 5.0.2 
    svelte: ^5.14.2 => 5.14.2 
    vite: ^6.0.1 => 6.0.3
```

**Comment by 360matt** [comment]
Fixed on Svelte v5.14.3
See https://github.com/sveltejs/vite-plugin-svelte/issues/1053

**Comment by dominikg** [comment]
leaving this open for now bc there is an adjacent bug in vite-plugin-svelte error handling regarding errors without a code property

**Comment by dominikg** [comment]
fixed in 4.0.4 and 5.0.3
